java-checks/src/test/files/checks/TwoLocksWaitNotifyCheck.java

```diff
@@ -0,0 +1,51 @@
+
+class Wait {
+
+  Object mon1;
+  Object mon2;
+
+  public Wait() {
+    synchronized (this.mon1) {
+      synchronized (this.mon2) {
+        this.mon2.wait();  // Noncompliant [[sc=9;ec=25;secondary=8,9]] {{Don't use "wait()" here; multiple locks are held.}}
+      }
+    }
+  }
+
+  public synchronized void method() {
+    synchronized (this.mon2) {
+      this.mon2.wait();  // Noncompliant [[sc=7;ec=23;secondary=15,16]] {{Don't use "wait()" here; multiple locks are held.}}
+    }
+  }
+}
+
+class Notify {
+
+  Object mon1;
+  Object mon2;
+
+  public Notify() {
+    synchronized (this.mon1) {
+      someotherStmt();
+      synchronized (this.mon2) {
+        this.mon2.notify();  // Noncompliant [[sc=9;ec=27;secondary=28,30]] {{Use "notifyAll()" here; multiple locks are held.}}
+      }
+    }
+  }
+
+  public synchronized void method() {
+    synchronized (this.mon2) {
+      this.mon2.notify();  // Noncompliant [[sc=7;ec=25;secondary=36,37]] {{Use "notifyAll()" here; multiple locks are held.}}
+    }
+  }
+
+  public synchronized void method() {
+    synchronized (this.mon2) {
+      this.mon2.notifyAll();  // Compliant
+    }
+  }
+
+  public void justForCoverage() {
```

Comment:
If this is useful, it is not documented enough, but I tend to think it should be removed.